### DISCUSSION TITLE
open .srs OMNIC

### DISCUSSION THREAD

**Discussion by dragonspirit93** [spectrochempy/spectrochempy#715] [Q&A] [answered]
Hello everyone, 
I'm new to this package, but I'm having some issue in opening series ThermoFisher data files (.srs) 
It seems that is not implemented yet, since the following warning appears on my console when I try to open one of my file. 

"WARNING | (UserWarning) Only implemented for rapidscan"

Is this implementation already being scheduled?

**Comment by atravert** [comment]
Hello,

Thanks you for your message.

There is indeed several types of series files (.srs) and for now only the rapid scan type is implemented, or at least for all types of rapid scan files I am aware of....

I can try to extend the reader for your type of file in few weeks, without warranty of success: these binary file formats are proprietary and reading them requires some reverse engineering.

If you are interested can you post a sample .srs file (preferably a few of them) and let us know the type if experiment to  which it corresponds ? I will look at it by mid August/end august.

Best regards
Arnaud

----- Mail d’origine -----
De: dragonspirit93 ***@***.***>
À: spectrochempy/spectrochempy ***@***.***>
Cc: Subscribed ***@***.***>
Envoyé: Wed, 02 Aug 2023 12:37:53 +0200 (CEST)
Objet: [spectrochempy/spectrochempy] open .srs OMNIC (Discussion #715)

Hello everyone,
I'm new to this package, but I'm having some issue in opening series ThermoFisher data files (.srs)
It seems that is not implemented yet, since the following warning appears on my console when I try to open one of my file.

"WARNING | (UserWarning) Only implemented for rapidscan"

Is this implementation already being scheduled?

**Comment by Micsyl** [comment]
Hello,

I am following up on this thread because I face the same issue (I am new to Spectrochempy and Python in general).
More specifically, to add to the first post, I tried and got the following:

`>>>X = scp.read('rapid_scan.srs')`
`NDDataset: [float64] V (shape: (y:643, x:4160))`

and

`>>>Y = scp.read('TGA_demo.srs')`
`WARNING: Only implemented for rapidscan`
`None`

Both files from the "spectrochempy_data/testdata/irdata/omnic_series" repository from Github.
I have the same issue with my own file, issued by Omnic. I suppose that if the issue is solved for the TGA_demo.srs, there are chances that it could work for mine.

If relevant, here are some information about my system:
Spectrochempy version: 0.4.10 and 0.6.8.dev13 (I tried both)
Omnic version: Omnic32 9.12.0.928

Thank you for your time and attention.

**Comment by atravert** [comment]
Hello @Micsyl 

Thank you for your message. There are several .srs format (TGA, kinetics, rapid-scans, .. and probably sub-formats with these categories). So the safest way to update the omnic reader is to use would be to use one of your sample file and comparing the output of spectrochempy with that of OMNIC. 
So, unless you are absolutely sure that your srs files have the same format as the TGA_demo.srs,  can, you post one of your sample file ?  I can work on the reader in the coming days,

Best regards
Arnaud.

**Comment by Micsyl** [comment]
Hello @atravert ,

Thank you for your answer. Obviously, I have no clue of the precise format of my file. The FTIR machine that I am using does not give much info about it and there is no Omnic reader that I can access; the machine only returns the chemical composition corresponding to the spectra.
 
Here is a link to get a sample file: https://we.tl/t-WqAHwrLAHS
I hope you can work that out.

Best regards, have a nice day

**Reply by atravert** [reply]
Thank you. Here is the omnic display from your file:

![image](https://github.com/spectrochempy/spectrochempy/assets/10098314/4853b213-2d7d-48ba-b551-34cff4000d06)

![image](https://github.com/spectrochempy/spectrochempy/assets/10098314/9990936c-8efd-442d-838f-0ca0c4983b72)
The data collection type is indicated as "High speed real time". I will try first to update the reader so as to extract the the set of spectra and let you know...
Best regards

**Reply by atravert** [reply]
Hello @Micsyl,

I have updated the reader, which is now able to read the set of spectra of your file (and the background if needed). Before pushing this update I'd like to add a test so as to ensure that the reader works fine at each update of spectrochempy or of external libraries used by spectrochempy. 
Do you authorize us to use your file for these tests?

Thanks,
Arnaud

**Reply by Micsyl** [reply]
Hello @atravert,

That is very good news, thank you for your efforts.
Of course, you can use the file for testing. It was a blank measurement of N2 only, for information.

Have a nice day

**Comment by atravert** [comment, marked as answer]
Hello @Micsyl,

The update has been pushed to the development branch few days ago (see [here](https://www.spectrochempy.fr/latest/gettingstarted/install/install_sources.html) for the installation). You can use the following to read your sample file which has been added to the [spectrochempy_data](https://github.com/spectrochempy/spectrochempy_data) repository.

```python
# to read the spectra of a high speed series
a = scp.read_srs("irdata/omnic_series/high_speed.srs")

# to read the background: 
a = scp.read_srs("irdata/omnic_series/high_speed.srs", return_bg=True)
```
Best regards
Arnaud.
